client/wildcard/src/components/Tooltip/Tooltip.tsx

```diff
@@ -101,26 +133,33 @@ export const Tooltip: FC<TooltipProps> = props => {
             popoverElement?.removeEventListener('pointerenter', handlePointerEnter)
             popoverElement?.removeEventListener('pointerleave', handlePointerLeave)
         }
-    }, [open])
+    }, [isOpen, setOpen])
 
     const handleOpenChange = (event: PopoverOpenEvent): void => {
         switch (event.reason) {
-            case PopoverOpenEventReason.Esc:
+            case PopoverOpenEventReason.Esc: {
+                setOpen({ isOpen: event.isOpen, reason: TooltipOpenEventReason.Esc })
+                return
+            }
             case PopoverOpenEventReason.ClickOutside: {
-                setOpen(false)
+                setOpen({ isOpen: event.isOpen, reason: TooltipOpenEventReason.ClickOutside })
+                return
             }
         }
     }
 
-    const isOpen = useDebounce(open, 100)
+    const tooltipId = `tooltip-${useId()}`
+    const isTooltipOpen = useDebounce(isOpen, 100)
```

Comment:
To make it clear from the variable name, the difference from `isOpen`.

```suggestion
    const isOpenDebounced = useDebounce(isOpen, 100)
```